### PR TITLE
First batch of rules for the Pyramid framework

### PR DESCRIPTION
**Pull request by p4p3r**
This PR contains rules for the Pyramid Python framework that cover:
- Cookie flag security
- CSRF configuration (global and per-view)
- SQLi related to SQLAlchemy usage (similar to existing rules for SQLAlchemy)
- XSS related to a few cases of Response object creation without using a template renderer

### CODE REVIEW

**Review by 0xDC0DE — APPROVED** [review]
I think [this](https://github.com/returntocorp/semgrep-rules/pull/2025/files#diff-016244119020973a7b2457d11f1b228f3e90dfe055fdeb348f290bd525805a59) rule can be split into two rules that can have fix options (but I'm unsure about the `**params` option)

You could have a rule for the missing httpflag, and one for the httpflag being set to false.
Example here: https://github.com/returntocorp/semgrep-rules/tree/develop/python/jinja2/security/audit

**Comment by 0xDC0DE** [comment]
Same argument for [python/pyramid/audit/authtkt-cookie-secure.yaml](https://github.com/returntocorp/semgrep-rules/pull/2025/files#diff-d80e994da8034265493ae4bb814233f3b28fa1e7a9ecd0bb26c259ac58179d39)

**Comment by 0xDC0DE** [comment]
And this one python/pyramid/audit/set-cookie-httponly.yaml

**Comment by 0xDC0DE** [comment]
But really, I love that you already wrote a lot of fixes for the other rules!

**Comment by 0xDC0DE** [comment]
I think you can add a fix to this rule as well: python/pyramid/audit/set-cookie-samesite.yaml

File: python/pyramid/audit/set-cookie-samesite.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                  $RESPONSE = $REQUEST.response
+                  ...
+          - pattern-inside: |
+              def $VIEW(...):
+                  ...
+                  $RESPONSE = pyramid.httpexceptions.HTTPFound(...)
+                  ...
+      - pattern-not: $RESPONSE.set_cookie(..., samesite='Lax', ...)
+      - pattern-not: $RESPONSE.set_cookie(..., **$PARAMS)
+      - pattern: $RESPONSE.set_cookie(..., samesite=$SAMESITE, ...)
```
Comment (0xDC0DE) [inline python/pyramid/audit/set-cookie-samesite.yaml]: It looks like this rule only checks for the case where `samesite` is set, but incorrectly. Is the default value secure?

**Review by 0xDC0DE — APPROVED** [review]

File: python/pyramid/audit/set-cookie-samesite.yaml (hunk 1 above)
Comment (p4p3r) [reply python/pyramid/audit/set-cookie-samesite.yaml]: Fixed in 4ea7a800e8b951fceca2d6876a6e51c060a17fbe

**Comment by p4p3r** [comment]
@pieterdc1 I split all the cookie flags rules into default-value/unsafe-value and added fixes to all the rules.

**Review by 0xDC0DE — APPROVED** [review]
I like your naming scheme for the split rules, I will be using that from now on.
`...-unsafe-default` and `...-unsafe-value`.

**Pull request merged**